README.md

```diff
@@ -54,33 +65,94 @@ This section describes the contributed pseudo debugger types and their support t
+
+**Workaround**: You can generate a HEX file for the program download, and the ELF file for debug purposes only. The following steps are required if you build a [CSolution](https://open-cmsis-pack.github.io/cmsis-toolbox/build-overview/)-based application with the [CMSIS-Toolbox](https://open-cmsis-pack.github.io/cmsis-toolbox/):
+
+1. Edit the `*.cproject.yml` file(s) of your application.
+1. Modify the [`output:type:`](https://open-cmsis-pack.github.io/cmsis-toolbox/YML-Input-Format/#output) node to generate both an `elf` and a `hex` file:
+```
+  output:
+    type:
+      - elf
+      - hex  
+```
+1. Build the solution.
+1. Keep the default configuration's `program` setting as is.
+```
+"program": "${command:cmsis-csolution.getBinaryFile}",
+```
+1. Modify the default debug configuration's `initCommands` list, so that the `load` command gets the relative path to the generated HEX file.
+```
+            "initCommands": [
+                "load ./relative/path/to/my/application.hex",
+                "break main"
+            ],
+```
 
-**Note**: The generated default debug configuration uses `JLinkGDBServer` as `target`.`server` setting. The executable with this name has slightly differing behavior depending on your host platform. It launches a GUI-less server on Linux and macOS. Whereas a GDB server with GUI is launched on Windows®. Please change the value to `JLinkGDBServerCL` to suppress the GUI on Windows.
+The debugger then loads the debug information from the ELF file. But uses the HEX file instead of the ELF file for program download.
 
-In addition this extension contributes a debug configuration resolver which automatically fills the following gaps during debug launch:
+### `arm-none-eabi-gdb` requires DWARF5 debug information
 
-- Adds/extends the `target`.`serverParameters` list of `JLinkGDBServer`/`JLinkGDBServerCL` command line arguments:
-  - Appends `--port` and the corresponding `port` value if `target`.`port` is set.
+`arm-none-eabi-gdb` generates the following warnings when I debug ELF files with [DWARF](https://dwarfstd.org/) debug information of standard version 4 and earlier. And the debug illusion seems to be broken in many places.<br>
+```
+warning: (Internal error: pc 0x8006a18 in read in CU, but not in symtab.)
+```
 
-## Known Limitations
+**Possible Reason**: `arm-none-eabi-gdb` works best with DWARF debug information of standard version 5.
 
-- Requires ELF files built with GCC and DWARF5 debug information to operate seamlessly.
-- The shipped pyOCD version accepts the new command line option `--cbuild-run`. But only extracts device and DFP names.
+**Solution**: Make sure to build your application ELF file with DWARF version 5 debug information. Please refer to your toolchain's user reference manual. This may require updates to all build tools like compiler, assembler, and linker.
```

Comment:
Just a comment: this is weak and does not help the user to do the right thing. We should include exact instructions for AC6 e.g. in the next version!